### PR TITLE
feat(auto-migrate): add labelSelector and readyReplicas fields to FTC path definitions

### PR DESCRIPTION
**Pull request by gary-lgy**

### CODE REVIEW

File: pkg/apis/core/v1alpha1/types_federatedtypeconfig.go [hunk 1]
```diff
@@ -104,6 +104,11 @@ type FederatedTypeConfigSpec struct {
 }
 
 type PathDefinition struct {
+	// Path to a metav1.LabelSelector field that selects the replicas for this object.
+	// E.g. `spec.selector` for Deployment and ReplicaSet.
+	// +optional
+	LabelSelector string `json:"labelSelector,omitempty"`
```
[the remark this answers is not in the document]
Comment (gary-lgy) [reply pkg/apis/core/v1alpha1/types_federatedtypeconfig.go:110]: Why is this field different from `replicas` and `availableReplicas`?

[the remark this answers is not in the document]
Comment (SOF3) [reply pkg/apis/core/v1alpha1/types_federatedtypeconfig.go:110]: `availableReplicas` is currently unused. `replicas` is just a general quantity that can be used to quantify anything, not just pods. But for `LabelSelector`, our usages only interpret it as pods.

[the remark this answers is not in the document]
Comment (gary-lgy) [reply pkg/apis/core/v1alpha1/types_federatedtypeconfig.go:110]: You could make the argument in the other direction: label selector is used to select replicas that are not necessarily pods. Furthermore, our current usages of `replicas` only interpret it as pods. I don't see how `replicas` and `labelSelector` are fundamentally different.

[the remark this answers is not in the document]
Comment (gary-lgy) [reply pkg/apis/core/v1alpha1/types_federatedtypeconfig.go:110]: In the context of auto migration, all these fields only apply to pods. But that doesn't mean these can't be interpreted in a broader sense by other components. In other words I see this stricter interpretation by auto migration as a limitation of a specific controller, which shouldn't affect the semantics of the API which can be consumed elsewhere

[the remark this answers is not in the document]
Comment (gary-lgy) [reply pkg/apis/core/v1alpha1/types_federatedtypeconfig.go:110]: @SOF3 bump

[the remark this answers is not in the document]
Comment (SOF3) [reply pkg/apis/core/v1alpha1/types_federatedtypeconfig.go:110]: Alright, maybe we can define this label selector as "selector for the replicable object populated in the member cluster as a consequence of the propagated object", and then define the auto migration feature as "the replication target of the propagated object must be exactly v1/Pod".

**Review by SOF3 — APPROVED** [review]

**Pull request merged**